lib/wasix/src/state/linker/module_handle.rs

```diff
@@ -0,0 +1,133 @@
+use derive_more::Debug;
+
+// IDs for special module handles
+// Not public to ensure they are only used through the safe API.
+// TODO: These should probably move to the C bindings at some point
+
+/// Constant representing the RTLD_DEFAULT flag (0) - for searching in all loaded objects
+const RAW_MODULE_HANDLE_RTLD_DEFAULT: u32 = 0;
+/// Constant representing the main module ID (1)
+const RAW_MODULE_HANDLE_MAIN: u32 = 1;
+/// Constant representing an invalid module handle (u32::MAX)
+const RAW_MODULE_HANDLE_INVALID: u32 = u32::MAX;
+
+/// A handle of a dynamic loaded shared object returned by dlopen
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum ModuleHandleWithFlags {
+    /// An invalid module handle
+    Invalid,
```

Comment:
This case doesn't really make sense except for a very special case (that of recursively loading module trees), and there's no point in polluting the code with it.